### ISSUE TITLE
The document version selection menu should support ‘preliminary’

### ISSUE THREAD

**Issue by SataQiu** [kubernetes/website#14691]
**This is a Bug Report**

**Problem:**
At present, there may be some problems with our document version management.
The correspondence between the document version and the code branch is as follows:
```
v1.14 -> master
v1.13 -> release-1.13
v1.12 -> release-1.12
...
```
I think the correct correspondence should be:
```
preliminary -> master
v1.14 -> release-1.14
v1.13 -> release-1.13
v1.12 -> release-1.12
...
```
The document version selection menu should support ‘preliminary’.

![图片](https://user-images.githubusercontent.com/9354727/58781733-44e4fd80-860f-11e9-8bab-3db78ba4c6c7.png)

**Proposed Solution:**

**Page to Update:**
https://kubernetes.io/...

/cc @zacharysarah  @markthink @chenrui333

**Comment by sftim** [comment]
If we wanted to publish documentation for versions we hadn't yet released, we could do that by publishing the `dev-1.15` or whatever branch.

Is that useful to do?

**Comment by SataQiu** [comment]
Yes, it's worth considering.  @sftim 
We should consider whether `dev` branches need to be shown, and how to do that.

**Comment by chenrui333** [comment]
@SataQiu I think the version management list should be like this:

```
v1.14 -> master (bug fix)
v1.15 -> dev-1.15 (feature)
v1.13 -> release-1.13
v1.12 -> release-1.12
...
```

while master branch is more for bug fix, dev-1.15 is for new feature contents.

**Comment by kbhawkey** [comment]
Can you provide an example of when publishing the dev-1.15 branch docs would be useful?
Perhaps there is some confusion about the choice of names for the branches and the name of what is currently published to kubernetes.io/docs (1.14 docs + all changes not targeted specifically for the next release, 1.15). There is a window of transition between the snapshot of a release and the next release.
